subsys/bluetooth/host/hci_core.c

```diff
@@ -4684,10 +4685,11 @@ static void le_per_adv_sync_lost(struct net_buf *buf)
 	/* Deleting before callback, so the caller will be able to restart
 	 * sync in the callback
 	 */
+	cb = per_adv_sync->cb;
 	per_adv_sync_delete(per_adv_sync);
 
-	if (per_adv_sync->cb && per_adv_sync->cb->term) {
-		per_adv_sync->cb->term(per_adv_sync, &term_info);
+	if (cb && cb->term) {
+		cb->term(per_adv_sync, &term_info);
```

Comment:
I think it would be cleaner to just no clean up the callback pointer in the PA sync object, as that doesn't really do anything functionality-wise. Once the flags have been cleared in per_adv_sync_delete, the object will be reused for future `new` PA objects, and the callbacks will always be overwritten, so there's actually no need to clear the callback pointer on delete, aside from it being symbolic. 

I would suggest just to remove the line that clears the callback in `per_adv_sync_delete`.